### PR TITLE
DOC updating spatial merge docs with argument details

### PR DESCRIPTION
**Pull request by lwasser**
This is a followup PR to 
https://github.com/geopandas/geopandas/issues/732

I am updating the documentation for gpd.sjoin() to be a bit more explicit. 

Please bare with me as this is my first actual contribution to an OS repo (i've submitted issues but never contributed content). I've directly edited the `.rst` files and welcome comments wherever you'd like changes. 

Once i understand how to best suggest changes I will update other parts of the docs as i create lessons for my python class! :)

### CODE REVIEW

File: doc/source/aggregation_with_dissolve.rst [hunk 1]
```diff
@@ -10,9 +10,9 @@
 Aggregation with dissolve
 =============================
 
-It is often the case that we find ourselves working with spatial data that is more granular than we need. For example, we might have data on sub-national units, but we're actually interested in studying patterns at the level of countries.
+Spatial data are often more granular than we need. For example, we might have data on sub-national units, but we're actually interested in studying patterns at the level of countries.
```
Comment (jorisvandenbossche) [inline doc/source/aggregation_with_dissolve.rst]: This are nice improvements for readability!

File: doc/source/aggregation_with_dissolve.rst [hunk 2]
```diff
@@ -35,7 +35,7 @@ First, let's look at the most simple case where we just want continent shapes an
 
     continents.head()
 
-If we are interested in aggregate populations, however, we can pass different functions to the ``dissolve`` method to aggregate populations:
+If we are interested in aggregate populations, however, we can pass different functions to the ``dissolve`` method to aggregate populations using the ``aggfun =`` argument:
```
Comment (jorisvandenbossche) [inline doc/source/aggregation_with_dissolve.rst]: aggfun -> aggfunc

File: doc/source/aggregation_with_dissolve.rst [hunk 3]
```diff
@@ -57,3 +57,14 @@ If we are interested in aggregate populations, however, we can pass different fu
 
 .. toctree::
    :maxdepth: 2
+
+### Dissolve Arguments
```
Comment (jorisvandenbossche) [inline doc/source/aggregation_with_dissolve.rst]: In rst files, the header should in this case be like:

```
Dissolve Arguments
~~~~~~~~~~~~~~~~~~
```

You can see how the other titles are in this file (rst is a bit more complex than markdown .. and this is one of the differences)

File: doc/source/aggregation_with_dissolve.rst [hunk 4]
```diff
@@ -57,3 +57,14 @@ If we are interested in aggregate populations, however, we can pass different fu
...
+
+The ``aggfun =`` argument defaults to 'first' which means that the first row of attributes values found in the dissolve routine will be applied to the resultant dissolved geodataframe.
```
Comment (jorisvandenbossche) [inline doc/source/aggregation_with_dissolve.rst]: aggfun -> aggfunc

Maybe "applied" -> "assigned" ? 
It sounds a bit strange to "apply values"

File: doc/source/aggregation_with_dissolve.rst [hunk 5]
```diff
@@ -57,3 +57,14 @@ If we are interested in aggregate populations, however, we can pass different fu
...
+However it also accepts other summary statistic options as allowed by ``pandas.groupby()`` including:
+
+* first
```
Comment (jorisvandenbossche) [inline doc/source/aggregation_with_dissolve.rst]: I would make this `'first'` (to make clear it is a string)

File: doc/source/mergingdata.rst [hunk 6]
```diff
@@ -72,6 +72,24 @@ In a Spatial Join, two geometry objects are merged based on their spatial relati
    cities_with_country.head()
 
 
-The ``op`` options determines the type of join operation to apply. ``op`` can be set to "intersects", "within" or "contains" (these are all equivalent when joining points to polygons, but differ when joining polygons to other polygons or lines).
+### Sjoin Arguments:
```
Comment (jorisvandenbossche) [inline doc/source/mergingdata.rst]: similar comment here about the title

File: doc/source/mergingdata.rst [hunk 7]
```diff
@@ -72,6 +72,24 @@ In a Spatial Join, two geometry objects are merged based on their spatial relati
...
+
+``sjoin.()`` has two core arguments: ``how`` and ``op``.
+
+#### op
+The ```op`` argument specifies how ``geopandas`` decides whether or not to join the attributes of one object to another. There are three different join options as follows:
+
+* `intersects`: The attributes will be joined if the boundary and interior of the object intersect in any way with the boundary and/or interior of the other object.
+* `within`: The attributes will be join if the object’s boundary and interior intersect *only* with the interior of the other object (not its boundary or exterior).
```
Comment (jorisvandenbossche) [inline doc/source/mergingdata.rst]: join -> joined

File: doc/source/mergingdata.rst [hunk 8]
```diff
@@ -72,6 +72,24 @@ In a Spatial Join, two geometry objects are merged based on their spatial relati
...
+* `contains`: The attributes will be joined if the object’s interior contains the boundary and interior of the other object and their boundaries do not touch at all.
+
+<a href ="http://toblerity.org/shapely/manual.html#binary-predicates" target = "_blank">You can read more about each join type in the Shapely documentation.</a>
```
Comment (jorisvandenbossche) [inline doc/source/mergingdata.rst]: The rst syntax for this is 

```
You can read more about each join type in the `Shapely documentation <http://toblerity.org/shapely/manual.html#binary-predicates>`__ 
```

(see http://www.sphinx-doc.org/en/stable/rest.html#external-links)

File: doc/source/aggregation_with_dissolve.rst (hunk 4 above)
Comment (lwasser) [reply doc/source/aggregation_with_dissolve.rst]: assigned is much better. thank you @jorisvandenbossche

File: doc/source/aggregation_with_dissolve.rst (hunk 3 above)
Comment (lwasser) [reply doc/source/aggregation_with_dissolve.rst]: my mistake! i noticed some differences from markdown and should have done my research PRIOR to submitted this PR. thank you!

File: doc/source/aggregation_with_dissolve.rst (hunk 5 above)
Comment (lwasser) [reply doc/source/aggregation_with_dissolve.rst]: Hopefully I understand correctly. I made this change:
```
* 'first'
* 'last'
* 'min'
* 'max'
* 'sum'
```

**Comment by lwasser** [comment]
Thank you for your patience with the syntax errors @jorisvandenbossche .
One question - it seems ~~~ is equivalent to heading 3. For each argument I then just bolded  what i  might set as a H4 as i didn't see an H4 in the rst docs. Like so:

```
Sjoin Arguments
~~~~~~~~~~~~~~~~

``sjoin.()`` has two core arguments: ``how`` and ``op``.

**op**
```
Please let me know if this seems reasonable. If not, i'll happily change it.

**Comment by lwasser** [comment]
I have fixed items as listed above @jorisvandenbossche and am happy to make any more changes as you see fit. Thank you again.

**Comment by lwasser** [comment]
hey @jorisvandenbossche  i am checkin in on this PR again. it does look like travis is failing but i'm not sure it is because of the docs. if it is please let me know. i'm also posting because we ran into a lack of documentation issue with dissolve during a recent workshop. i'd like to update the dissolve docs too but want your input on adding to this PR vs a new PR for that. please let me know. If this one could be merged i would happily start a fresh pr.

**Comment by jorisvandenbossche** [comment]
@lwasser Thanks a lot! (and looking forward to other mentioned doc updates :))

**Comment by lwasser** [comment]
oh wonderful!! i've been traveling and just got back to see this. glad i could help @jorisvandenbossche ... i will work on updating the dissolve docs as well. thank you for merging this!!

**Pull request merged**